src/components/Problems/Problems.tsx

```diff
@@ -58,6 +57,63 @@ interface Series {
   semester: number
 }
 
+const Problem: FC<{
+  problem: Problem
+  registered: boolean
+  commentCount: number
+  setDisplaySideContent: Dispatch<
+    SetStateAction<{
+      type: string
+      problemId: number
+      problemNumber: number
+    }>
+  >
+  canRegister: boolean
+}> = ({problem, registered, commentCount, setDisplaySideContent, canRegister}) => {
+  const handleDiscussionButtonClick = () => {
+    setDisplaySideContent((prevState) => {
+      if (prevState.type === 'discussion' && prevState.problemId === problem.id) {
+        return {type: '', problemId: -1, problemNumber: -1}
+      } else {
+        return {type: 'discussion', problemId: problem.id, problemNumber: problem.order}
+      }
+    })
+  }
+  const handleUploadClick = () => {
+    if (canRegister) {

```

Comment:
toto som upravil tiez, kym nemame nejaku lepsiu funkcionalitu, ked clovek odovzdava a nie je zaregistrovany,- dal som, nech to krici hned. ale prave tu mozno vieme radsej uplne disablenut button a ukazat nejaky tooltip na hoveri napriklad, ze sa ma registrovat do serie. aspon bude vediet, ze hej, odovzdavat by sa mohlo dat, ale musi nieco spravit